vpc/tool/container2/setup_container_linux.go

```diff
@@ -745,25 +727,59 @@ func deleteFromIPv6BPFMap(ctx context.Context, assignment *vpcapi.AssignIPRespon
 	defer unix.Close(int(fd))
 	runtime.KeepAlive(path)
 
+	key := ipv6key(assignment)
 	uba := bpfAttrMapOpElem{
 		mapFd: uint32(fd),
-		key:   uint64(uintptr(unsafe.Pointer(&buf[0]))),
+		key:   uint64(uintptr(unsafe.Pointer(&key[0]))),
 	}
 	_, _, err = unix.Syscall(
 		unix.SYS_BPF,
 		unix.BPF_MAP_DELETE_ELEM,
 		uintptr(unsafe.Pointer(&uba)),
 		unsafe.Sizeof(uba),
 	)
-	if err != unix.ENOENT && err != 0 {
+	if err == unix.ENOENT {
+		err := errors.Wrap(err, "Unable to delete element for ipv6 map (notfound)")
+		logger.G(ctx).WithError(err).WithField("ip", assignment.Ipv6Address.Address.Address).WithField("classid", assignment.ClassId).WithField("vlanid", assignment.VlanId).Error("Element not found")
+		return err
+	} else if err != 0 {
 		logger.G(ctx).WithError(errors.Wrapf(err, "Unable to delete element for map with file descriptor %d", fd)).Error()
-		err2 := errors.Wrap(err, "Unable to delete element for ipv4 map")
+		err2 := errors.Wrap(err, "Unable to delete element for ipv6 map")
 		return err2
 	}
 	runtime.KeepAlive(uba)
 	return nil
 }
 
+func ipv6key(assignment *vpcapi.AssignIPResponseV3) []byte {
+	if assignment.Ipv6Address == nil {
+		return nil
+	}
+	buf := make([]byte, 20)
+	binary.LittleEndian.PutUint16(buf, uint16(assignment.VlanId))
+	ip := net.ParseIP(assignment.Ipv6Address.Address.Address).To16()
+	if len(ip) != 16 {
+		panic("Length of IP is not 16")
+	}
+	copy(buf[4:], ip)
+	return buf
+}
+
+func ipv4key(assignment *vpcapi.AssignIPResponseV3) []byte {
+	if assignment.Ipv4Address == nil {
+		return nil
+	}
+	buf := make([]byte, 8)
+
+	binary.LittleEndian.PutUint16(buf, uint16(assignment.VlanId))
+	ip := net.ParseIP(assignment.Ipv4Address.Address.Address).To4()
+	if len(ip) != 4 {
+		panic("Length of IP is not 4")
+	}
+	copy(buf[4:], ip)
```

Comment:
Out of curiosity - What do we do with the first 4 bytes of this buffer eventually?

Reply:
VLAN ID / sg